pkg/core/consensus/selection/selector.go

```diff
@@ -138,30 +150,28 @@ func (s *Selector) IncreaseTimeOut() {
 	s.timeout = s.timeout * 2
 }
 
-func (s *Selector) publishBestEvent() error {
+func (s *Selector) sendBestEvent() error {
+	var bestEvent message.Score
 	s.eventPlayer.Pause(s.scoreID)
-	buf := new(bytes.Buffer)
 	s.lock.RLock()
-	bestEvent := s.bestEvent
+	bestEvent = s.bestEvent
 	s.lock.RUnlock()
+
 	// If we had no best event, we should send an empty hash
-	if bestEvent == nil {
-		s.signer.SendInternally(topics.BestScore, emptyScore[:], buf, s.ID())
-	} else {
-		s.signer.SendInternally(topics.BestScore, bestEvent.VoteHash, buf, s.ID())
+	if bestEvent.IsEmpty() {
+		bestEvent = s.signer.Compose(emptyScoreFactory{}).(message.Score)
```

Comment:
It appears that the `emptyScoreFactory` composes a `ScoreProposal`, which causes another panic here.